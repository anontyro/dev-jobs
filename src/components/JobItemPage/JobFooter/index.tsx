import React from "react";
import styled from "@emotion/styled";
import JobQueryData from "../../../interfaces/jobQueryData";

const JobFooterContainer = styled.div`
  display: flex;
  height: 5rem;
  .job-footer-container {
    width: 60%;
    margin: 0 auto;
    display: flex;
  }
  .job-footer-body {
    display: flex;
    flex-direction: column;
    flex-grow: 1;
    justify-content: center;
    text-align: left;
    h3 {
      margin: 0;
    }
    p {
      margin: 0.3rem 0;
    }
  }
  .job-footer-button {
  }
  @media (max-width: 768px) {
  }
  @media (max-width: 425px) {
    height: 4rem;
    .job-footer-container {
      width: 95%;
    }
    .job-footer-body {
      display: none;
    }
    .job-footer-button {
      width: 100%;
      button {
        width: 100%;
      }
    }
  }
`;

interface JobPageProps {
  job: JobQueryData;
}
const JobFooter: React.FC<JobPageProps> = ({ job }) => (
  <JobFooterContainer className="footer job">
    <div className="job-footer-container">
      <div className="job-footer-body">
        <h3 className="bold">{job.title}</h3>
        <p className="card-text-muted">{job.company.name}</p>
      </div>
      <div className="flex-center-items job-footer-button">
        <button className="button default">Apply Now</button>
      </div>
    </div>
  </JobFooterContainer>
);

export default JobFooter;
