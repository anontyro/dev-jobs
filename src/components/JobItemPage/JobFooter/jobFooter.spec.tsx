import React from "react";
import renderer from "react-test-renderer";
import JobFooter from ".";
import JobQueryData from "../../../interfaces/jobQueryData";

describe("JobFooter", () => {
  it("renders correctly", () => {
    const job = {
      title: "test job",
      company: {
        name: "the company",
      },
    } as unknown as JobQueryData;
    const tree = renderer.create(<JobFooter job={job} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
