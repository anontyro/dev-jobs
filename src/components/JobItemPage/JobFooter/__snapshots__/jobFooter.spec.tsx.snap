// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobFooter renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-yve4ho{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;height:5rem;}.css-yve4ho .job-footer-container{width:60%;margin:0 auto;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}.css-yve4ho .job-footer-body{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;text-align:left;}.css-yve4ho .job-footer-body h3{margin:0;}.css-yve4ho .job-footer-body p{margin:0.3rem 0;}@media (max-width: 425px){.css-yve4ho{height:4rem;}.css-yve4ho .job-footer-container{width:95%;}.css-yve4ho .job-footer-body{display:none;}.css-yve4ho .job-footer-button{width:100%;}.css-yve4ho .job-footer-button button{width:100%;}}",
      }
    }
    data-emotion="css yve4ho"
  />,
  <div
    className="footer job css-yve4ho"
  >
    <div
      className="job-footer-container"
    >
      <div
        className="job-footer-body"
      >
        <h3
          className="bold"
        >
          test job
        </h3>
        <p
          className="card-text-muted"
        >
          the company
        </p>
      </div>
      <div
        className="flex-center-items job-footer-button"
      >
        <button
          className="button default"
        >
          Apply Now
        </button>
      </div>
    </div>
  </div>,
]
`;
